geode-core/src/main/java/org/apache/geode/management/internal/operation/OperationHistoryManager.java

```diff
@@ -90,6 +95,27 @@ private static boolean isExpired(long expirationTime, OperationState<?, ?> opera
     return operationEnd.getTime() <= expirationTime;
   }
 
+  private OperationState<?, ?> validateLocator(OperationState<?, ?> operationState) {
+    if (isLocatorOffline(operationState)) {
+      operationState.setOperationEnd(new Date(), null,
+          new RuntimeException("Locator that initiated the Rest API operation is offline: "
+              + operationState.getLocator()));
+    }
+
+    return operationState;
+  }
+
+  private boolean isLocatorOffline(OperationState operationState) {
+    if (operationState.getOperationEnd() == null
+        && (operationState.getLocator() != null)
+        && cache.getMyId().toString().compareTo(operationState.getLocator()) != 0
```

Comment:
Does it need to be compared? can it be changed to "equals"....

Reply:
Good point!